subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/radio/radio_df.c

```diff
@@ -262,9 +274,20 @@ void radio_df_ant_switch_spacing_set_4us(void)
 	radio_df_ant_switch_spacing_set(RADIO_DFECTRL1_TSWITCHSPACING_4us);
 }
 
-void radio_df_ant_switch_pattern_set(uint8_t pattern)
+void radio_df_ant_switch_pattern_set(uint8_t *patterns, uint8_t len)
 {
-	NRF_RADIO->SWITCHPATTERN = pattern;
+	/* DFE extension in radio uses:
+	 * - SWITCHPATTER[0] for idle period (PDU Tx/Rx),
+	 * - SWITCHPATTER[1] for guard and reference period,
+	 * - SWITCHPATTER[2] and following for switch-sampling slots.
+	 * Due to that in SWITCHPATTER[0] there is stored a pattern provided by
+	 * DTS property dfe_pdu_ant. This limits number of supported antenna
+	 * switch patterns by one.
+	 */
+	NRF_RADIO->SWITCHPATTERN = DT_PROP(RADIO, dfe_pdu_ant);
```

Comment:
I'm not able to match this with the comment above, shouldn`t SWITCHPATTERN be indexed?

Reply:
This is correct. The SWITCHPATTER is kind of a pointer to underlying buffer. Every write moves the pointer to next free memory location. When read returns number of written elements. If want to start writing from beginning you should write to CLEARPATTERN register to before.

Reply:
Ok, thank you.